contracts/Everlong.sol

```diff
@@ -307,35 +299,99 @@ contract Everlong is IEverlong {
     // │ Rebalancing                                             │
     // ╰─────────────────────────────────────────────────────────╯
 
-    // TODO: Handle case where rebalancing would exceed gas limit
-    // TODO: Handle when Hyperdrive has insufficient liquidity.
+    // NOTE: Errors from hyperdrive are not handled. The keeper must configure
+    //       the correct parameters to avoid issues with insufficient liquidity
+    //       and running out of gas from mature position closures.
     //
     /// @notice Rebalance the everlong portfolio by closing mature positions
     ///         and using the proceeds over target idle to open new positions.
-    function rebalance() public override {
+    function rebalance() external onlyAdmin {
+        _rebalance(
+            IEverlong.RebalanceOptions({
+                spendingOverride: 0,
+                minOutput: 0,
+                minVaultSharePrice: 0,
+                positionClosureLimit: 0,
+                extraData: ""
+            })
+        );
+    }
+
+    // NOTE: Errors from hyperdrive are not handled. The keeper must configure
+    //       the correct parameters to avoid issues with insufficient liquidity
+    //       and running out of gas from mature position closures.
+    //
+    /// @notice Rebalance the everlong portfolio by closing mature positions
+    ///         and using the proceeds over target idle to open new positions.
+    /// @param _options Options to control the rebalance behavior.
+    function rebalance(
+        IEverlong.RebalanceOptions memory _options
+    ) external onlyAdmin {
+        _rebalance(_options);
```

Comment:
What's the point of having the internal and external variants of this function? I don't see `_rebalance` being called anywhere else. 

If we decide to keep the rebalance function with the default options (honestly not my preference -- I'd rather we just have one), we can just make this public.